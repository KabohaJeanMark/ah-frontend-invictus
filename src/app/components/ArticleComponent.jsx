import React from 'react';
import { WithContext as ReactTags } from 'react-tag-input';
import '../../assets/styles/article.scss';
import Navbar from '../components/Navbar';

const KeyCodes = {
  comma: 188,
  enter: 13
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];

const ArticlePresentationComponent = ({ handleSubmit, handleChange, handleAddTag, handleDeleteTag, tags }) => {
  return (
    <>
    <Navbar />
      <div className="create-article">
        <div className="row">
          <div className="col-md-2" />
          <div className="col-md-8">
            <div className="card">
              <form onSubmit={handleSubmit}>
                <div className="form-group">
                  <label className="sr-only">Title</label>
                  <textarea
                    className="title-input"
                    type="text"
                    name="title"
                    id="title-input"
                    placeholder="Title"
                    required
                    onChange={(e) => handleChange(e)}/>
                </div>
                <div className="form-group">
                  <textarea
                    className="description-input"
                    type="text"
                    name="description"
                    id="description-input"
                    placeholder="Description"
                    required
                    onChange={(e) => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <textarea
                    className="body-input"
                    type="text"
                    name="body"
                    id="body-input"
                    placeholder="Body"
                    required
                    onChange={e => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="tags">Tags</label>
                  <ReactTags
                    classNames="tags"
                    maxLength = "42"
                    inline
                    tags={tags}
                    handleAddition={handleAddTag}
                    handleDelete={handleDeleteTag}
                    delimiters={delimiters}
                  />
                </div>
                <button type="submit" className="btn postarticle-btn btn-outline-primary">Post Article</button>
              </form>
            </div>
          </div>
          <div className="col-md-2" />
        </div>
      </div>
    </>
  );
};

export default ArticlePresentationComponent;
