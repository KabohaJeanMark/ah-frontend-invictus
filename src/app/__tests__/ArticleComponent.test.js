import React from 'react';
import { shallow } from 'enzyme';
import ArticleComponent from '../components/ArticleComponent';

describe('Article Component', () => {
  it('renders without errors', () => {
    const component = shallow(<ArticleComponent />);
    const wrapper = component.find('.create-article');
    expect(wrapper.length).toBe(1);
  });
  it('handles events without errors', () => {
    const props = {
      handleChange: jest.fn()
    };
    const wrapper = shallow(<ArticleComponent {...props} />);
    wrapper.find('#title-input').simulate('change');
    wrapper.find('#description-input').simulate('change');
    wrapper.find('#body-input').simulate('change');
  });
});
