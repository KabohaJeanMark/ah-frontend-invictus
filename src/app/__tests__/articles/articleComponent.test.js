import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ArticleComponent from '../../components/articles/getArticleComponent';

const props = {
  article: {
    title: 'Example Title',
    body: 'Some text',
    author: 'Test',
    description: 'Test'
  },
  auth: {
    username: "sanya"
  }
};

describe('article component', () => {
  let wrapper = shallow(<ArticleComponent {...props} />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('article component again', () => {
  props.article.id = 5;
  props.article.tagList = ['edna'];
  let wrapper = shallow(<ArticleComponent {...props} />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
