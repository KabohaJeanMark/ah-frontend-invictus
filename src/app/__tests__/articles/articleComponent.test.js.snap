// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`article component matches snapshot 1`] = `
<Fragment>
  <div>
    <Connect(NavbarComponent) />
=======
exports[`article component again matches snapshot 1`] = `
<div>
  <Connect(NavbarComponent) />
  <div
    className="container"
  >
>>>>>>> feat (Share Articles): User should be able to share articles on social media.
    <div
      className="container"
    >
<<<<<<< HEAD
=======
      <h1
        className="article-title"
      >
        Example Title
      </h1>
      <div>
        <Connect(RateArticle)
          articleId={5}
        />
        <_default
          author="Test"
          body="Some text"
          description="Test"
          id={5}
          tagList={
            Array [
              "edna",
            ]
          }
          title="Example Title"
        />
      </div>
      <div
        className="article-desc"
      >
        Test
      </div>
      <div>
        Test
      </div>
      <img
        alt=""
        className="card-image"
        src="https://picsum.photos/600/300"
      />
      <div
        className="article-body"
      >
        Some text
      </div>
>>>>>>> feat (Share Articles): User should be able to share articles on social media.
      <div
        className="article-main"
      >
        <div
          className="row"
        >
          <div
            className="col-md-10"
          >
            <h1
              className="article-title"
            >
              Example Title
            </h1>
          </div>
        </div>
        <div
          className="article-desc"
        >
          Test
        </div>
        <div>
          Test
        </div>
        <img
          alt=""
          className="card-image"
          src="https://picsum.photos/600/300"
        />
        <div
          className="article-body"
        >
          Some text
        </div>
        <div
          className="like-dislike"
        >
          <i
            className="fa fa-thumbs-up"
            role="presentation"
            style={
              Object {
                "color": "blue",
                "marginRight": "10px",
              }
            }
          />
          <i
            className="fa fa-thumbs-down"
            role="presentation"
            style={
              Object {
                "color": "red",
              }
            }
          />
        </div>
        <div
          className="all-tags"
        >
          <hr />
          <ul>
            <li
              className="tag"
              key="edna"
            >
              edna
            </li>
          </ul>
          <div>
            <Connect(CommentContainer) />
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`article component matches snapshot 1`] = `
<div>
  <Connect(NavbarComponent) />
  <div
    className="container"
  >
    <div
      className="article-main"
    >
      <h1
        className="article-title"
      >
        Example Title
      </h1>
      <div
        className="article-desc"
      >
        Test
      </div>
      <div>
        Test
      </div>
      <img
        alt=""
        className="card-image"
        src="https://picsum.photos/600/300"
      />
      <div
        className="article-body"
      >
        Some text
      </div>
      <div
        className="like-dislike"
      >
        <i
          className="fa fa-thumbs-up"
          role="presentation"
          style={
            Object {
              "color": "blue",
              "marginRight": "10px",
            }
          }
        />
        <i
          className="fa fa-thumbs-down"
          role="presentation"
          style={
            Object {
              "color": "red",
            }
          }
        />
      </div>
      <div
        className="all-tags"
      >
        <hr />
        <ul />
        <div>
          <Connect(CommentContainer) />
        </div>
      </div>
    </div>
  </div>
</div>
`;
